server/src/routers/converterRouter.js

```diff
@@ -0,0 +1,63 @@
+const fs = require('fs')
+const path = require('path')
+const express = require('express')
+const multer = require('multer')
+const upload = multer()
+const YAML = require('yamljs')
+const router = new express.Router()
+
+const { mapping } = require('../constants/config')
+
+router.use(express.json())
+
+var Converter = require('api-spec-converter')
+
+//Endpoint for conversion
+router.post('/api/specification/:format/:convertTo', upload.single('file'), async (req, res) => {
+    var file = req.file,
+        {format, convertTo} = req.params,
+        {syntax, toFile} = req.query
+        buffer = Buffer.from(file.buffer),
+        origFile = buffer.toString(),
+        options = {syntax: 'yaml', order: 'openapi'}
+
+    // Check if the conversion is possible
+    if(mapping[format] && mapping[format].includes(convertTo)) {
+
+      //Parse JSON / YAML
+      if(path.parse(file.originalname).ext === '.json') {
+        origFile = JSON.parse(origFile)
+      }
+      else {
+        origFile = YAML.parse(origFile)
+      }
+
+      //Conversion using Lucy's Converter
+      Converter.convert({
+          from: format,
+          to: convertTo,
+          source: origFile,
+      }, (err, result) => {
+          if(err) {
+              res.status(500).send({'message': err})
+          }
+          else {
+            result = syntax === 'yaml' ? result.stringify(options) : result
+            if(toFile) {
+              fs.writeFileSync(`./${path.parse(file.originalname).name}.${syntax ? syntax : 'json'}`, syntax === 'yaml' ? result : JSON.stringify(result))
```

Comment:
What happens if we get two requests with the same file name?

Reply:
Adding deletion of the file after sending response.

Reply:
I've added the deletion of file after the response has been sent. This must solve the problem.

Reply:
The question would still remain, if the two requests were received at the same time, right?

Reply:
Right, now the buffer is being used instead of saving the file to the disk. That should fix the issue.